subprojects/execution/src/test/groovy/org/gradle/internal/execution/fingerprint/impl/DefaultInputFingerprinterTest.groovy

```diff
@@ -70,6 +70,7 @@ class DefaultInputFingerprinterTest extends Specification {
         1 * valueSnapshotter.snapshot(input) >> inputSnapshot
         1 * snapshotter.snapshot(fileInput) >> fileInputSnapshotResult
         _ * fileInputSnapshotResult.fileTreeOnly >> false
+        _ * fileInputSnapshotResult.containsArchiveTrees() >> false
```

Comment:
Shall we add a test about the property showing up among inputs requiring empty check if this is true?

Reply:
Added.